From 7cac90ab1de93ab9852b4a1f8d9413cfd63ca39b Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Tue, 18 Sep 2018 03:46:25 +0300
Subject: [PATCH 1/1] Treat repo adding as same URI as a warning and remove it
 before adding it again as a potentially new URI. This should hopefully
 resolve an issue in solbuild where we attempt to do main / stable image
 creation, it fails as a result of the URI already existing (as the default),
 then proceeds to do baselayout and system.base component install failing.

---
 pisi/api.py | 27 +++++++++++----------------
 1 file changed, 11 insertions(+), 16 deletions(-)

diff --git a/pisi/api.py b/pisi/api.py
index 24ec323..528cdd8 100644
--- a/pisi/api.py
+++ b/pisi/api.py
@@ -800,18 +800,19 @@ def add_repo(name, indexuri, at = None):
 
     if repodb.has_repo_url(new_indexuri, only_active = False):
         repo = repodb.get_repo_by_url(new_indexuri)
-        raise pisi.Error(_('Repo already present with name %s and same URL.') % repo)
-    else:
-        if repodb.has_repo(name):
-            repodb.remove_repo(name)
+        ctx.ui.warning(_('Repo already present with name %s and same URL. Removing first.') % repo)
+        repodb.remove_repo(repo)
 
-        if new_indexuri != indexuri:
-            ctx.ui.warning("Legacy repo found. Attempting rewrite.")
+    if repodb.has_repo(name):
+        repodb.remove_repo(name)
 
-        repo = pisi.db.repodb.Repo(pisi.uri.URI(new_indexuri))
-        repodb.add_repo(name, repo, at = at)
-        pisi.db.flush_caches()
-        ctx.ui.info(_('Repo %s added to system.') % name)
+    if new_indexuri != indexuri:
+        ctx.ui.warning("Legacy repo found. Attempting rewrite.")
+
+    repo = pisi.db.repodb.Repo(pisi.uri.URI(new_indexuri))
+    repodb.add_repo(name, repo, at = at)
+    pisi.db.flush_caches()
+    ctx.ui.info(_('Repo %s added to system.') % name)
 
 @locked
 def remove_repo(name):
@@ -861,12 +862,6 @@ def __update_repo(repo, force=False):
 
         pisi.db.historydb.HistoryDB().update_repo(repo, repouri, "update")
         repodb.check_distribution(repo)
-
-        try:
-            index.check_signature(repouri, repo)
-        except pisi.file.NoSignatureFound, e:
-            ctx.ui.warning(e)
-
         ctx.ui.info(_('Package database updated.'))
     else:
         raise pisi.Error(_('No repository named %s found.') % repo)
-- 
2.18.0

